src/compiler/checker.ts

```diff
@@ -12075,6 +12077,12 @@ namespace ts {
             return !!(getObjectFlags(type) & ObjectFlags.Reference) && (<TypeReference>type).target === globalArrayType;
         }
 
+        function isPromiseLikeType(type: Type): boolean {
+            const globalPromiseType = getGlobalPromiseType(/*reportErrors*/ false); // this is only called from the language service, so don't report errors if the promise type doesn't exist

```

Comment:
Why do you shadow the `globalPromiseType` here created in `initializeTypeChecker`?